R/summarise_draws.R

```diff
@@ -118,62 +174,59 @@ summarise_draws.draws <- function(x, ..., .args = list()) {
 
   # it is more efficient to repair and transform objects for all variables
   # at once instead of doing it within the loop for each variable separately
-  if (ndraws(x) == 0L) {
+  if (ndraws(.x) == 0L) {
     return(empty_draws_summary())
   }
-  x <- repair_draws(x)
-  x <- as_draws_array(x)
-  variables <- variables(x)
+  .x <- repair_draws(.x)
+  .x <- as_draws_array(.x)
+  variables_x <- variables(.x)
   
-  if (!length(variables)) {
+  if (!length(variables_x)) {
     warning_no_call(
       "The draws object contained no variables with unreserved names. ",
       "No summaries were computed."
     )
     return(tibble::tibble(character()))
   }
   
-  create_summary_list <- function(x, v) {
-    draws <- drop_dims_or_classes(x[, , v], dims = 3, reset_class = FALSE)
-    args <- c(list(draws), .args)
-    v_summary <- named_list(names(funs))
-    for (m in names(funs)) {
-      v_summary[[m]] <- do.call(funs[[m]], args)
+  if (cores == 1) {
+    out <- summarise_draws_helper(.x, funs, .args)
+  } else {
+    .x <- .x[ , , !(dimnames(.x)$variable %in% reserved_variables())]
```

Comment:
`variables.draws_array()` removes reserved variable names internally. So I think this can be simplified to: 

```r
x <- x[ , , variables(x)]
```

Reply:
So just checking, if true then presumably we can just remove this entire line?

Reply:
If you want to be 100% sure reserved variable names are dropped then I think you still need it: 

```r
x <- example_draws() 
variables(x)[1] <- ".log_weight" # reserved name
print(x, max_variables = 1) # includes variable with reserved name 
print(x[,, variables(x)], max_variables = 1)   # doesn't include variable with reserved name
```

But now I'm wondering why we don't prevent printing reserved variables but want to prevent summarizing them (maybe because we don't usually care about inference on something like `.log_weight`?). @paul-buerkner what do you think?

Reply:
I've implemented @jgabry's simplification here.  I won't resolve this conversation pending comments on @jgabry's comment above.

Reply:
Let's move this discussion to a separate issue.  @jgabry do you want to open it?

Reply:
Here's the issue: https://github.com/stan-dev/posterior/issues/146